### ISSUE TITLE
[ISSUE] Importing GIF Image Sequence will show empty frames

### ISSUE THREAD

**Issue by Kaikogf** [pencil2d/pencil#903]
I imported 37 PNG image files from a simple animation into Pencil2D but all frames were empty/blank after imported.

The number of frames where imported correctly. Still those frames do not allow to draw on them, all strokes are hiding after releasing  the pen pressure...

Version Pencil2D build 23Feb2018
win 8.1 x 64bits

EDITED: 03April 2018 > PGN files are OK. GIF files shows this issue

**Comment by Kaikogf** [comment]
@chchwy 
Not sure if I made a mistake at the time about this issue with PNG files, but today (build v. 02Apr2018) PNG are OK, only GIF files are the ones to show this issue.
I am going to edit the main title

**Comment by Jose-Moreno** [comment]
@Kaikogf Can you please confirm if this still happens for you on the latest nightly build?

**Comment by Kaikogf** [comment]
@Jose-Moreno 
Tested with version20Jun2018

It brings a new issue. Now pencil2D is unable to load any GIF image into the timeline. The following error appears:

![pencilgif](https://user-images.githubusercontent.com/28032931/42164202-3455331a-7dfd-11e8-9c86-3bc6b022e009.jpg)

It doesn't have to do with the number of GIF images. 
But tested the **_File > Import > Image_**, and it is working fine !
:+1:

**Comment by Jose-Moreno** [comment]
@Kaikogf To bring animated GIF files you always had to use File > Import > Image as a single GIF file is still considered an individual image.

As for the error please be more specific, did it appear when you attempted to import image sequences with the GIF format?

**Comment by Kaikogf** [comment]
@Jose-Moreno 
The error is when trying to import GIF image sequence. Even importing one single image with the import image sequence doesn't work

**Comment by CandyFace** [comment]
I don't think we've ever allowed importing multiple gifs into a project. I found an old build from 2017 and it wasn't possible in that one either.

Not sure what the proper action is here... A gif is already a bunch of images, so the question is whether it makes sense to be able to import multiple.

**Comment by Jose-Moreno** [comment]
@CandyFace I believe the issue is more about having both features working properly while trying to import any GIF file. From a personal perspective I feel there is a design flaw that we should fix while were at it. The still image importer should not be allower to import Animated GIF files (multiple frames) but rather the "image sequence importer" is the one that should be allowed to bring the Animated GIF files, that way we can use the frame spacing property.

This would also keep the "still image importer" focused in it's appropriate functionality which is to bring any still image format (GIF included) into Pencil2D. What could be added as well so we don't get an error necessarily is that IF we try to import an animated GIF using the "still image importer" it will import just the first frame instead of the whole sequence :slightly_smiling_face:

**Comment by CandyFace** [comment]
@Jose-Moreno 
Hmm... so you're proposing that if you import a gif as a still image, you only want it to consider the first image in the stack? 

I'm not sure there's a point to allowing importing a gif and only taking the first image in the stack. I'm thinking that we should simply remove gif from still image import instead and only allow it through import sequence. I agree that it doesn't make sense to import gif with the import image option.

**Comment by Jose-Moreno** [comment]
@CandyFace Separating the import Animated GIF option sounds good to me :+1: And while we're at it, maybe separating the export isn't a bad idea either. Many people are always confused as to why we have to choose the movie option.

Regarding the Image Sequence importer possibility, there are users who have asked how to import animated GIF's only after attempting to do it through the "image sequence" import option. I also suggested that because I've required many times to import the GIF with a different spacing between the imported frames to compensate for the difference between the original animation speed and the editor FPS (e.g The GIF is 8FPS but the timeline is 24FPS) and I felt the frame spacing importer option could potentially help solve this without having to make an "interpret footage" feature or placing the imported frames manually.

The "import the first frame" solution was mentioned hypothetically as if the GIF importer had been changed from between the import features to avoid giving an error message to the user. As a user one normally would expect things to work themselves up to a certain degree, instead of giving you an error when the rest of the options are working properly. It is confusing.

With that said I'm not really sure what to do here.
My main concern is: are we supposed to deprecate an entire workflow choice just because we think it's obsolete? If Pencil2D has been implicitly supporting this feature ever since it's inception, but it has never worked properly as you noted before, then the problem lies in the fact that It has never worked, and it should be fixed to work as expected with sibling formats.

**Comment by CandyFace** [comment]
>  If Pencil2D has been implicitly supporting this feature ever since it's inception, but it has never worked properly as you noted before, then the problem lies in the fact that It has never worked, and it should be fixed to work as expected with sibling formats.

Importing gif as a sequence has never been possible so there's no workflow to deprecate? It's true that you've been able to export gif via the movie exporter and import gif via "import image" for some time, but I don't believe we should keep this workflow regardless of people being used to doing it that way.

We are still in development phase, so nothing is set in stone and everything is therefore subject to change, which the users should expect.

conclusion of this:
+ Exporting/importing gif should be available via new options.
  - Importing will look similar to image sequence dialog
  - Export will look similar to export movie but with gif as the only format.

**Comment by Jose-Moreno** [comment]
@CandyFace I don't really mind personally. As long as the users can import / export animated GIF's I think there won't be complains. So I'm agreeing here to separate the GIF option on it's own dialog window and to remove any trace of a GIF "image sequence" import.

It's mostly a UI change, as the functionality will remain the same. I think it will be ok though, to specify very clearly in the release notes and the future manual that users can but only import individual GIF files (one frame images) or animated GIF files (multi-frame) but can't bring individual GIF image sequences, because seriously until recently I always thought you could do that, I just never had the need to do so and if it weren't for this report I'd probably kept thinking that heh :smile:
